src/swift/SwiftAPIViewCore/Sources/Models/APIViewModel.swift

```diff
@@ -0,0 +1,313 @@
+
+    /// The navigation tokens to display
+    var navigation: [NavigationToken]
+
+    /// Node-based representation of the Swift package
+    var model: PackageModel
+
+    /// Current indentation level
+    private var indentLevel = 0
+
+    /// Whether indentation is needed
+    private var needsIndent = false
+
+    /// Number of spaces to indent per level
+    let indentSpaces = 4
+
+    /// Access modifier to expose via APIView
+    static let publicModifiers: [AccessLevelModifier] = [.public, .open]
+
+    /// Tracks assigned definition IDs so they can be linked
+    private var definitionIds = Set<String>()
+
+    /// Returns the text-based representation of all tokens
+    var text: String {
+        return tokens.map { $0.text }.joined()
+    }
+    
+    // MARK: Initializers
+
+    init(name: String, packageName: String, versionString: String, statements: [Statement]) {
+        self.name = name
+        self.versionString = versionString
+        self.packageName = packageName
+        navigation = [NavigationToken]()
+        tokens = [Token]()
+        model = PackageModel(name: packageName, statements: statements)
+        self.tokenize(apiview: self)
+        model.navigationTokenize(apiview: self)
+    }
+
+    // MARK: Codable
+
+    enum CodingKeys: String, CodingKey {
+        case name = "Name"
+        case tokens = "Tokens"
+        case language = "Language"
+        case packageName = "PackageName"
+        case navigation = "Navigation"
+        case versionString = "VersionString"
+    }
+
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(name, forKey: .name)
+        try container.encode(packageName, forKey: .packageName)
+        try container.encode(language, forKey: .language)
+        try container.encode(tokens, forKey: .tokens)
+        try container.encode(navigation, forKey: .navigation)
+        try container.encode(versionString, forKey: .versionString)
+    }
+
+    func tokenize(apiview a: APIViewModel) {
+        // Renders the APIView "preamble"
+        let bundle = Bundle(for: Swift.type(of: self))
+        let versionKey = "CFBundleShortVersionString"
+        let apiViewVersion = bundle.object(forInfoDictionaryKey: versionKey) as? String ?? "Unknown"
+        a.text("Package parsed using Swift APIView (version \(apiViewVersion))")
+        a.newline()
+        a.blankLines(set: 2)
+        model.tokenize(apiview: a)
+    }
+
+    // MARK: Token Emitters
+    func add(token: Token) {
+        self.tokens.append(token)
+    }
+
+    func add(token: NavigationToken) {
+        self.navigation.append(token)
+    }
+
+    func text(_ text: String, definitionId: String? = nil) {
+        checkIndent()
+        let item = Token(definitionId: definitionId, navigateToId: nil, value: text, kind: .text)
+        // TODO: Add cross-language definition ID
+        // if add_cross_language_id:
```

Comment:
Is this just for Swift or across languages?

Reply:
I just left comment here so I know where to add it. Right now only Python has this implemented.